app/src/main/java/io/horizontalsystems/xrateskit/demo/CoinsInteractor.kt

```diff
@@ -54,6 +58,29 @@ class CoinsInteractor(private val ratesManager: RatesManager) {
         }
     }
 
+    fun getTopList(currency: String, shownSize: Int){
+        if (topListDisposable == null || topListDisposable?.isDisposed == true) {
+            ratesManager.topList(currency, shownSize)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    it.forEach { priceInfo ->
+                        Log.i("CoinsInteractor", "top list ${priceInfo.coin} ${priceInfo.price}")
+                    }
+                    topListDisposable?.dispose()
+                    //fetch second page
+                    if (shownSize == 0) {
+                        getTopList(currency, 50)
```

Comment:
getTopList(currency, 50).  can "50 " be set as const_value ?

Reply:
Here I just simulate requests. In app this number will come from adapter.